### PR TITLE
🏗  Lint for $ in animation-name

### PR DESCRIPTION
**Pull request by alanorozco**
Will error:

```diff
  const JSS = {
    foo: {
+ // The animation name in property animationName should start with $ (e.g. $foo).
+ // This scopes it to a keyframes rule present in this module.
      animationName: 'bar',
+ // ----------------^
    },
  };
```

```diff
  const JSS = {
    foo: {
+ // The animation name in property animation should start with $.
+ // This scopes it to a keyframes rule present in this module.
      animation: 'bar linear 0.5s',
+ // ------------^
    },
  };
```

### CODE REVIEW

**Comment by samouri** [comment]
Two thoughts:
1. Do we ever ever reference animation names from a different file, e.g. from ampshared?
2. My personal preference for a rule like this is not to autofix. I personally have a workflow where I exclusively run lint with the `--fix` option, and this could just as easily hide an issue as fix it

**Review by samouri — APPROVED** [review]
Had two small comments, overall lgtm

**Comment by alanorozco** [comment]
1. No, they're always set in their same CSS file, so it wouldn't break scoping.
2. Fair enough, removed fix function.

File: build-system/eslint-rules/jss-animation-name.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+        return;
+      }
+      // Could be {key: val} identifier or {'key': val} string
+      const keyName = node.key.name || node.key.value;
+      if (keyName !== 'animationName' && keyName !== 'animation') {
+        return;
+      }
+      if (typeof node.value.value !== 'string') {
+        context.report({
+          loc: node.value.loc,
```
Comment (jridgewell) [inline build-system/eslint-rules/jss-animation-name.js:47]:
```suggestion
          node: node.value,
```

File: build-system/eslint-rules/jss-animation-name.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+          message: `Use string literals for ${keyName} values.`,
+        });
+        return;
+      }
+      if (
+        (keyName === 'animationName' && !/^\$/.test(node.value.value)) ||
+        (keyName === 'animation' && !/(^| )\$/.test(node.value.value))
+      ) {
+        context.report({
+          loc: node.value.loc,
```
Comment (jridgewell) [inline build-system/eslint-rules/jss-animation-name.js:57]:
```suggestion
          node: node.value,
```

**Review by jridgewell — APPROVED** [review]

**Pull request merged**